okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -356,10 +351,20 @@ public void run() {
       try {
         // Read until the underlying socket closes.
         while (frameReader.nextFrame(this)) {
```

Comment:
After the latest update, there are only 2 cases:

1. Server closes TCP connection. nextFrame will return false and we exit the loop
2. We close the socket, which causes nextFrame return false immediately.